src/irmin-pack/unix/gc.ml

```diff
@@ -167,14 +138,22 @@ module Make (Args : Gc_args.S) = struct
     Commit_store.purge_lru t.commit;
     Ok ()
 
-  let unlink_all { root; generation; _ } =
+  let unlink_all { root; generation; _ } chunk_idx_gced =
     let result =
       let open Result_syntax in
-      (* Unlink previous suffix. *)
-      let suffix =
-        Irmin_pack.Layout.V4.suffix_chunk ~root ~chunk_idx:(generation - 1)
+      (* Unlink suffix chunks *)
+      let* () =
+        let rec unlink idx =
```

Comment:
nitpick: you could use `List.iter`

Reply:
I wanted to preserve the result reporting of this function which is why I wrote it this way. Is there a nice way to iterate a list that does that for results (I couldn't find anything with some brief searching)?

Reply:
sorry, you are right, I looked over it too quickly. 

One thing we could do is define the missing `List.iter_with_results` or something in `irmin/import.ml` for instance, instead of defining it here. Wdyt?

Reply:
I think it's a good idea if we have other instances. I'd want to spend some time thinking about the design of a function (should it live on `List` or `Result`, for instance), so maybe for another PR. I can add a comment/TODO here to remind us.

Reply:
You could use a function that combines the list and result monad:
```ml
let rec iter_result f = function
  | [] -> Ok ()
  | hd :: tl -> Result.bind (f hd) (fun () -> iter_result f tl)
```

```ml
val iter_result : ('a -> (unit, 'b) result) -> 'a list -> (unit, 'b) result
```

Reply:
Changed in [`ad270b8` (#2126)](https://github.com/mirage/irmin/pull/2126/commits/ad270b846b65bd57ab670a8703c7c470a8d1ee1d). Much nicer!